### PR TITLE
S23 create user profile page

### PR DESCRIPTION
**Pull request by Riaboshtan**
I've got a lot of troubles:
1) I can't show user first_name
2) I don't know how to change password  (at the same page)
3) I don't know how to save changes

### CODE REVIEW

File: apps/kindergarten/templates/show_kindergarten.html [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+{% endblock %}
+
+
+{% block main %}
+
+<div class="example2">
+    <img src='http://www.freyarchitekten.com/wp-content/uploads/2017/02/Passivhauskindergarten_02.jpeg' class="example_beauty">
+    <span> <h2>{{ object.name }}</h2>
+           <h6>{{ object.street }} , {{ object.address}}</h6>
+           <a href="http://127.0.0.1:8000/register_child/"> <input class="button24" type="submit" value="Подати заяву" ></a>
```
Comment (aodarc) [inline apps/kindergarten/templates/show_kindergarten.html:17]: use {% url 'name of url' %} tag

File: apps/user_profile/models.py [hunk 2]
```diff
@@ -2,30 +2,32 @@
...
 
+from apps.location.models import City
 from core.models.helpers import image_path
 
 
 class UserProfile(models.Model):
     SEX_CHOICES = (
-        ('F', 'female'),
-        ('M', 'male')
+        ('Жіноча', 'female'),
```
Comment (aodarc) [inline apps/user_profile/models.py:11]: its true, why did you change it?

Comment (aodarc) [reply apps/user_profile/models.py:11]: In Django templates you can use the "get_FOO_display()" method, that will return the readable alias for the field, where 'FOO' is the name of the field.

File: apps/user_profile/models.py [hunk 3]
```diff
@@ -2,30 +2,32 @@
...
+        ('Чоловіча', 'male')
     )
 
-    avatar = models.ImageField(
+    '''avatar = models.ImageField(
         upload_to=image_path,
         blank=True
     )
     address = models.CharField(
```
Comment (aodarc) [inline apps/user_profile/models.py:19]: do not use ''' for making comments
also
Those fields should be left

File: apps/kindergarten/views.py [hunk 4]
```diff
@@ -1 +1,19 @@
...
+from apps.kindergarten.models import (
+    Kindergarten,
+    Group
+)
+from apps.kindergarten.templates import *
+
+from django.views.generic.detail import DetailView
+
+class KinderView(DetailView):
+    template_name = 'show_kindergarten.html'
```
Comment (aodarc) [inline apps/kindergarten/views.py:11]: change dir

File: apps/user_profile/models.py [hunk 5]
```diff
@@ -37,6 +39,34 @@ class UserProfile(models.Model):
         verbose_name='Номер телефону'
     )
 
+    city = models.ForeignKey(
+        to=City,
+        default=1
```
Comment (aodarc) [inline apps/user_profile/models.py:44]: don't use default value for FK, never

File: apps/user_profile/models.py [hunk 6]
```diff
@@ -37,6 +39,34 @@ class UserProfile(models.Model):
...
+    )
+
+    street = models.CharField(
+        verbose_name='Вулиця',
+        max_length=100,
+        default='',
```
Comment (aodarc) [inline apps/user_profile/models.py:50]: change default value to None
also
null=True

File: apps/user_profile/models.py [hunk 7]
```diff
@@ -37,6 +39,34 @@ class UserProfile(models.Model):
...
+    )
+
+    house = models.PositiveIntegerField(
+        verbose_name='Будинок',
+        default=0,
```
Comment (aodarc) [inline apps/user_profile/models.py:55]: same

File: apps/user_profile/models.py [hunk 8]
```diff
@@ -37,6 +39,34 @@ class UserProfile(models.Model):
...
+
+    house_symbol = models.CharField(
+        verbose_name='Символ будинку',
+        max_length=2,
+        blank=True,
+    )
+
+    flat = models.PositiveIntegerField(
+        verbose_name='Номер квартири',
+        default=0,
```
Comment (aodarc) [inline apps/user_profile/models.py:66]: same

File: apps/user_profile/static/css/password.css [hunk 9]
```diff
@@ -0,0 +1,10 @@
+.button1 {
```
Comment (aodarc) [inline apps/user_profile/static/css/password.css:1]: move it to css/show_kindergarten.css and delete this file

File: apps/user_profile/urls.py [hunk 10]
```diff
@@ -4,13 +4,19 @@
...
 
 urlpatterns = [
     url(r'^/', UserProfileDetailView.as_view()),
     url(r'^register/$', RegisterFormView.as_view()),
     url(r'^(?P<gender>[a-zA-Z]+)$', get_users_by_gender),
     url(r'^register/$', RegisterFormView.as_view(), name="register"),
-    url(r'^$', MainPageView.as_view(), name="main")
+    url(r'^$', MainPageView.as_view(), name="main"),
+    #url(r'^register_child$', child_form),
+    url(r'^profile_page/(?P<pk>[0-9]+)$', UserPageView.as_view(), name='user-page'),
```
Comment (aodarc) [inline apps/user_profile/urls.py:18]: it should be just pk, i mean without 'profile_page'

File: apps/user_profile/views.py [hunk 11]
```diff
@@ -42,3 +45,30 @@ class RegisterFormView(FormView):
...
+
+def change_password(request):
+    if request.method == 'POST':
+        form = PasswordChangeForm(request.user, request.POST)
+        if form.is_valid():
+            user = form.save()
+            update_session_auth_hash(request, user)  # Important!
+            return redirect('change_password')
+        else:
+            messages.error(request, 'Please correct the error below.')
```
Comment (aodarc) [inline apps/user_profile/views.py:69]: ops there will be error

**Review by aodarc — CHANGES_REQUESTED** [review]

File: apps/user_profile/models.py (hunk 5 above)
Comment (Riaboshtan) [reply apps/user_profile/models.py:44]: when i try to migrate without default value for any field, it shows 
(mvenv) riaboshtan@riaboshtan-E502NA:~/final_project$ python manage.py makemigrations
You are trying to add a non-nullable field 'city' to userprofile without a default; we can't do that (the database needs something to populate existing rows).
Please select a fix:
 1) Provide a one-off default now (will be set on all existing rows with a null value for this column)
 2) Quit, and let me add a default in models.py
Select an option:
